pkg/storage/memory/memory.go

```diff
@@ -242,14 +243,21 @@ func (s *MemoryBackend) ReadChanges(ctx context.Context, store string, filter st
 		return nil, nil, storage.ErrMismatchObjectType
 	}
 
-	var allChanges []*openfgav1.TupleChange
+	var allChanges []*TupleChangeRec
 	now := time.Now().UTC()
-	for _, change := range s.changes[store] {
-		if objectType == "" || (strings.HasPrefix(change.GetTupleKey().GetObject(), objectType+":")) {
-			if change.GetTimestamp().AsTime().After(now.Add(-horizonOffset)) {
+	for _, changeRec := range s.changes[store] {
```

Comment:
Why the need for this change? There are operations already being done against the timestamp like horizon trimming, the same can be done by converting the ULID to time and comparing across the timestamps, no need to have additional struct or memory for something that is already encoded into `TupleChange`. Number of total operations would be the same too since you'd only have to convert to Timestamp once to calculate against horizon and `from`.

Reply:
@elbuo8 `timestamp` didn't work for me because 
- in memory impl will use the same timestamp for the batch writes
- ulid generated from timestamp drops nanoseconds, in a test all write happen within the same millisecond, so paging becomes impossible

Reply:
@elbuo8 also we're planning to drop `in-mem` implementation by replacing it with SQLite in-mem #1989

Reply:
> by replacing it with SQLite in-mem

Been hearing that for a while :trollface: 

> ulid generated from timestamp drops nanoseconds, in a test all write happen within the same millisecond, so paging becomes impossible

Understandable 👍🏽